crate/src/lib.rs

```diff
@@ -214,12 +200,13 @@ pub enum Route {
 
 impl From<Url> for Route {
     fn from(url: Url) -> Self {
-        let mut path = url.path.into_iter();
+        let mut path = url.path().iter();
 
-        match path.next().as_deref() {
+        // `Option<&String>.as_defer() -> &str` doesn't work as with owned `String`
+        match path.next().map(|s| s.as_str()) {
```

Comment:
Maybe `match url.remaining_path_parts().as_slice() {` would be better.
See [this example](https://github.com/seed-rs/seed/blob/56538187545c10f32c4bcaa260b64c9c36a1e3b9/examples/url/src/lib.rs#L37) or [this one](https://github.com/seed-rs/seed/blob/56538187545c10f32c4bcaa260b64c9c36a1e3b9/examples/pages/src/page/admin/page/report.rs#L14).
(It may be changed a bit in the near future - see [this issue](https://github.com/seed-rs/seed/issues/433).)